packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js

```diff
@@ -229,54 +222,54 @@ export const EditFolderDialog = ({ onClose, folder, parentFolderId }) => {
                     </Stack>
                   </GridItem>
                 </Grid>
+              </ModalBody>
 
-                <VisuallyHidden>
-                  <button type="submit" tabIndex={-1} ref={submitButtonRef} name="hidden-submit">
-                    {formatMessage({ id: 'submit', defaultMessage: 'Submit' })}
-                  </button>
-                </VisuallyHidden>
-              </Form>
-            )}
-          </Formik>
-        </ModalBody>
-
-        <ModalFooter
-          startActions={
-            <Button onClick={() => onClose()} variant="tertiary" name="cancel">
-              {formatMessage({ id: 'cancel', defaultMessage: 'Cancel' })}
-            </Button>
-          }
-          endActions={
-            <Stack horizontal spacing={2}>
-              {isEditing && canUpdate && (
-                <Button
-                  type="button"
-                  variant="danger-light"
-                  onClick={() => setShowConfirmDialog(true)}
-                  name="delete"
-                >
-                  {formatMessage({
-                    id: 'modal.folder.create.delete',
-                    defaultMessage: 'Delete folder',
-                  })}
-                </Button>
-              )}
+              <ModalFooter
+                startActions={
+                  <Button
+                    onClick={() => onClose()}
+                    variant="tertiary"
+                    name="cancel"
+                    disabled={isEditFolderLoading}
+                  >
+                    {formatMessage({ id: 'cancel', defaultMessage: 'Cancel' })}
+                  </Button>
+                }
+                endActions={
+                  <Stack horizontal spacing={2}>
+                    {isEditing && canUpdate && (
+                      <Button
+                        type="button"
+                        variant="danger-light"
+                        onClick={() => setShowConfirmDialog(true)}
+                        name="delete"
+                        disabled={isEditFolderLoading}
```

Comment:
To disable the button is not really a good a11y practice, as the element doesn't receive any events anymore. Shouldn't this use `loading={isEditFolderDeleteLoading}`?

Reply:
I agree about not having 3 spinners. What do think about that:

- close: not disabled, no loading state, because clicking it does not involve any risk, because no request is being fired
- delete / submit: loading state on the button

Reply:
works for me!